pipeline/runtime/runtestsV2.go

```diff
@@ -425,6 +425,15 @@ func createSelectedTestFile(ctx context.Context, fs filesystem.FileSystem, stepI
 	tiConfig *tiCfg.Cfg, tmpFilepath string, envs map[string]string, runV2Config *api.RunTestsV2Config, filterFilePath string) error {
 	isManualExecution := instrumentation.IsManualExecution(tiConfig)
 	resp, isFilterFilePresent := getTestsSelection(ctx, fs, stepID, workspace, log, isManualExecution, tiConfig, envs, runV2Config)
+	if tiConfig.GetParseSavings() {
+		if isFilterFilePresent {
```

Comment:
If the filter file is present but is empty, we run all the tests, right?

Reply:
During full run we are not writing the filter file? I think the original design was different.
cc: @smjt-h please review.